src/WebUI/WorkLoads/WorkloadDetails.tsx

```diff
@@ -61,29 +68,75 @@ export interface IWorkLoadDetailsItem {
 export class WorkloadDetails extends BaseComponent<IWorkloadDetailsProperties, IWorkloadDetailsState> {
     constructor(props: IWorkloadDetailsProperties) {
         super(props, {});
+        this._podsStore = StoreManager.GetStore<PodsStore>(PodsStore);
+
+        this._podsStore.addListener(PodsEvents.PodsFetchedEvent, this._onPodsUpdated);
+
+        this._workloadsStore = StoreManager.GetStore<WorkloadsStore>(WorkloadsStore);
+
+        let item = props.item;
+        if (!props.item) {
+            const kubeService = KubeSummary.getKubeService();
+            ActionsCreatorManager.GetActionCreator<PodsActionsCreator>(PodsActionsCreator).getPods(kubeService);
+            const workloadsActionCreator = ActionsCreatorManager.GetActionCreator<WorkloadsActionsCreator>(WorkloadsActionsCreator);
+
+            const history = createBrowserHistory();
+            const queryParams = queryString.parse(history.location.search);
+            if (queryParams.type) {
+                const getItem = (getItemList: () => (V1ReplicaSetList | V1StatefulSetList | V1DaemonSetList | undefined), actionCreatorFunc: (k: IKubeService) => void, storeEvent: string) => {
+                    const searchItem = () => {
```

Comment:
Please add some descriptive comments as to what getItem and searchItem functions are doing. It's not very intuitive from the code